File: api/src/main/java/eu/suro/utils/BukkitUtils.java
package eu.suro.utils;

import lombok.experimental.UtilityClass;
import org.bukkit.Bukkit;
import org.bukkit.Color;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;
import org.bukkit.scheduler.BukkitRunnable;
import org.bukkit.scheduler.BukkitTask;

import javax.annotation.Nullable;
import java.util.Objects;

@UtilityClass
public class BukkitUtils {

    public static final String VERSION = Bukkit.getServer().getClass().getName().split("\\.")[3]; //версия пакетов

    public static final int MIN_COORD = -29999999;
    public static final int MAX_COORD = 29999999;


    /**
     * Остановить таск с try catch IGNORE<br>
     * (Иногда при остановлении task'ка, выдает Exception)
     * @param task таск
     */
    public static void cancelTask(@Nullable BukkitTask task) {
        if(task != null) {
            try {
                task.cancel();
            } catch(Exception ignore) {}
        }
    }

    /**
     * Остановить таск с try catch IGNORE<br>
     * (Иногда при остановлении task'ка, выдает Exception)
     * @param task таск
     */
    public static void cancelTask(@Nullable BukkitRunnable task) {
        if(task != null) {
            try {
                task.cancel();
            } catch(Exception ignore) {}
        }
    }

    /**
     * Остановить таск с try catch IGNORE<br>
     * (Иногда при остановлении task'ка, выдает Exception)
     * @param task таск
     */
    public static void cancelTask(@Nullable ScheduleBukkit task) {
        if(task != null) {
            try {
                task.cancel();
            } catch(Exception ignore) {}
        }
    }
}

File: paper/src/main/kotlin/eu/suro/utils/BukkitUtils.kt
package eu.suro.utils

import org.bukkit.Bukkit
import org.bukkit.scheduler.BukkitRunnable
import org.bukkit.scheduler.BukkitTask

class BukkitUtils {

    companion object{
        val VERSION : String = Bukkit.getServer()::class.java.name.split("\\.")[3]
        val MIN_COORD = -29999999
        val MAX_COORD = 29999999

        /**
         * Остановить таск с try catch IGNORE<br>
         * (Иногда при остановлении task'ка, выдает Exception)
         * @param task таск
         */
        fun cancelTask(task: BukkitTask?){
            if(task != null){
                try {
                    task.cancel()
                }catch (ignore: Exception) {}
            }
        }

        /**
         * Остановить таск с try catch IGNORE<br>
         * (Иногда при остановлении task'ка, выдает Exception)
         * @param task таск
         */
        fun cancelTask(task: BukkitRunnable?) {
            if (task != null) {
                try {
                    task.cancel()
                } catch (ignore: java.lang.Exception) {
                }
            }
        }

        /**
         * Остановить таск с try catch IGNORE<br>
         * (Иногда при остановлении task'ка, выдает Exception)
         * @param task таск
         */
        fun cancelTask(task: ScheduleBukkit?) {
            if (task != null) {
                try {
                    task.cancel()
                } catch (ignore: java.lang.Exception) {
                }
            }
        }
    }
}